src/mesh/Data.hpp

```diff
@@ -60,12 +63,46 @@ class Data {
   /// Returns a const reference to the gradient data values.
   const Eigen::MatrixXd &gradients() const;
 
+  /**
+   * @brief Creates and initialized _waveform
+   *
+   * Currently needed, because Waveform requests PtrData. Creating PtrData(this) in constructor of data leads to memory management problems.
+   *
+   * @todo try to solve this differently and let exchange initialize Waveform of data. PtrStorage should be sufficient for Waveform, PtrData is overkill.
+   */
+  void initializeWaveform(PtrData ptrToMe,
+                          int     waveformDegree = time::Time::DEFAULT_WAVEFORM_DEGREE);
+
   /// Returns a reference to the _sample.
   time::Sample &sample();
 
   /// Returns a const reference to the _sample.
   const time::Sample &sample() const;
 
+  /**
+   * @brief Samples _waveform at given time
+   *
+   * @param normalizedDt Time where the sampling inside the window happens. Only allows values between 0 and 1. 0 refers to the beginning of the window and 1 to the end.
+   * @return Value of _waveform at time normalizedDt.
+   */
+  Eigen::VectorXd sampleAtTime(double normalizedDt) const;
+
+  /**
+   * @brief Get the _interpolationDegree.
+   *
+   * @return int _interpolationDegree
+   */
+  int getInterpolationDegree() const;
```

Comment:
```suggestion
  int getWaveformDegree() const;
```
? why a different word

Reply:
Mainly legacy, because we had interpolation vs. extrapolation degree at some point in time. But yes: Let's just use waveform degree everywhere.